tests/graphics/figimage_test.py

```diff
@@ -0,0 +1,158 @@
+    assert sc.identical(fig.artists[key]._data, da)
+
+
+def test_update_not_2d_raises():
+    fig = FigImage()
+    with pytest.raises(ValueError, match="FigImage can only be used to plot 2-D data."):
+        fig.update(data_array(ndim=1), key='data1d')
+    with pytest.raises(ValueError, match="FigImage can only be used to plot 2-D data."):
+        fig.update(data_array(ndim=3), key='data3d')
+
+
+def test_create_with_node():
+    da = data_array(ndim=2)
+    fig = FigImage(input_node(da))
+    assert len(fig.artists) == 1
+    assert sc.identical(list(fig.artists.values())[0]._data, da)
+
+
+def test_create_with_bin_edges():
+    da = data_array(ndim=2, binedges=True)
+    fig = FigImage(input_node(da))
+    assert len(fig.artists) == 1
+    assert sc.identical(list(fig.artists.values())[0]._data, da)
+
+
+def test_create_with_only_one_bin_edge_coord():
+    da = data_array(ndim=2, binedges=True)
+    da.coords['xx'] = sc.midpoints(da.coords['xx'])
+    fig = FigImage(input_node(da))
+    assert len(fig.artists) == 1
+    assert sc.identical(list(fig.artists.values())[0]._data, da)
+
+
+def test_log_norm():
+    fig = FigImage(norm='log')
+    assert fig.colormapper.norm == 'log'
+
+
+def test_toggle_norm():
+    da = data_array(ndim=2)
+    fig = FigImage(input_node(da))
+    assert fig.colormapper.norm == 'linear'
+    fig.toggle_norm()
+    assert fig.colormapper.norm == 'log'
+
+
+def test_crop():
+    da = data_array(ndim=2, binedges=True)
+    fig = FigImage(input_node(da))
+    assert fig.canvas.xrange == (da.meta['xx'].min().value, da.meta['xx'].max().value)
+    assert fig.canvas.yrange == (da.meta['yy'].min().value, da.meta['yy'].max().value)
+    xmin = sc.scalar(2.1, unit='m')
+    xmax = sc.scalar(102.0, unit='m')
+    ymin = sc.scalar(5.5, unit='m')
+    ymax = sc.scalar(22.3, unit='m')
+    fig.crop(xx={'min': xmin, 'max': xmax}, yy={'min': ymin, 'max': ymax})
+    assert fig.canvas.xrange == (xmin.value, xmax.value)
+    assert fig.canvas.yrange == (ymin.value, ymax.value)
+
+
+def test_crop_no_variable():
+    da = data_array(ndim=2)
+    xmin = 2.1
+    xmax = 102.0
+    ymin = 5.5
+    ymax = 22.3
+    fig = FigImage(input_node(da),
+                   crop={
+                       'xx': {
+                           'min': xmin,
+                           'max': xmax
+                       },
+                       'yy': {
+                           'min': ymin,
+                           'max': ymax
+                       }
+                   })
+    assert fig.canvas.xrange == (xmin, xmax)
+    assert fig.canvas.yrange == (ymin, ymax)
+
+
+@pytest.mark.parametrize('ext', ['jpg', 'png', 'pdf', 'svg'])
+def test_save_to_disk(ext):
+    da = data_array(ndim=2)
+    fig = FigImage(input_node(da))
+    with tempfile.TemporaryDirectory() as path:
+        fname = os.path.join(path, f'plopp_fig2d.{ext}')
+        fig.save(filename=fname)
+        assert os.path.isfile(fname)
```

Comment:
Again, missing test for actual file type. If that is difficult, would it make sense to check if providing an unsupported extension raises?

Reply:
Added a test that raises for bad extension.